src/main/java/impl/ui/Parser.java

```diff
@@ -0,0 +1,221 @@
+package impl.ui;
+
+import exceptions.InvalidCommandException;
+import impl.storage.TaskList;
+import impl.interfaces.Deadlines;
+import impl.interfaces.Events;
+import impl.interfaces.Task;
+import impl.interfaces.ToDos;
+
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+
+public class Parser {
+    enum Command{
+        List,
+        Todo,
+        Deadline,
+        Event,
+        Mark,
+        Unmark,
+        Delete,
+        Find,
+        Error,
+        Unknown
+    }
+    TaskList list;
+    /**
+     * Initialises Parser.
+     *
+     * @param list TaskList used to store tasks.
+     */
+    public Parser(TaskList list){
+        this.list = list;
+    }
+
+    /**
+     * Handles all String input by the user.
+     * Creates/mark/unmark/delete Tasks based on input
+     *
+     * @param in Input String.
+     */
+    public void handleInput(String in){
+        Command cmd = Command.Error;
+        try {
+            cmd = Parser.checkCommand(in, list.size());
+        }
+        catch(Exception e){
+            System.out.println(e.getMessage());
+            cmd = Parser.Command.Error;
+        }
+
+        switch (cmd){
+            case List:
+                handleList();
+                break;
+            case Todo:
+                handleToDo(in);
+                break;
+            case Deadline:
+                handleDeadline(in);
+                break;
+            case Event:
+                handleEvent(in);
+                break;
+            case Mark:
+                handleMark(in);
+                break;
+            case Unmark:
+                handleUnmark(in);
+                break;
+            case Delete:
+                handleDelete(in);
+                break;
+            case Find:
+                handleFind(in);
+                break;
+            case Error:
+                System.out.println("Error! Please try again :(");
+                break;
+            case Unknown:
+                // Fall through as default
+            default:
+                System.out.println("Unknown Command Detected!");
+                break;
```

Comment:
I think you might need to modify the style for switch statements as
   switch() { 
   case "T" : 
'case' must be aligned with 'switch'